src/flib/pionfget_mod.F90.in

```diff
@@ -499,4 +490,204 @@ CONTAINS
     ierr = get_var_{DIMS}d_{TYPE} (File, vardesc%varid, ival)
 
   end function get_var_vdesc_{DIMS}d_{TYPE}
+
+! DIMS 0,1,2,3,4,5
+  subroutine Cstring2Fstring_{DIMS}d(cstr, cstr_sz, max_clen, fstr)
+    character(C_CHAR), intent(in) :: cstr(:)
+    ! The total size of cstr, the total number of chars in cstr
+    integer, intent(in) :: cstr_sz
+    ! The length of each string in cstr
+    integer, intent(in) :: max_clen
+    character(len=*), intent(out) :: fstr{DIMSTR}
+
+#if {DIMS} != 0
+    integer :: fstr_dim_sz({DIMS})
+#endif
+    integer :: max_nstrs, max_flen
+    integer :: cstr_idx
+    logical :: cstr_copied
+    integer :: i, j, k, m, n, q
+
+    character, parameter :: F_SPACE_CHAR = ' '
+
+    max_nstrs = cstr_sz / max_clen
+#if {DIMS} != 0
+    if(max_nstrs > size(fstr)) then
+      print *, "ERROR: The provided user buffer",&
+                  " does not have space for copying ", max_nstrs, "strings",&
+                  " (space available to copy only ", size(fstr), " strings)"
+      call piodie(__PIO_FILE__, __LINE__, "Insufficient user buffer space")
+    end if
+
+    do i=1,{DIMS}
+      fstr_dim_sz(i) = size(fstr,i)
+    enddo
+#else
+    if(max_nstrs > 1) then
+      print *, "ERROR: The provided user buffer",&
+                  " does not have space for copying ", max_nstrs, "strings",&
+                  " (space available to copy only 1 string)"
+      call piodie(__PIO_FILE__, __LINE__, "Insufficient user buffer space")
+    end if
+#endif
+    fstr = F_SPACE_CHAR
+    cstr_idx = 0
+#if {DIMS} == 0
+    max_flen= len(fstr)
+    cstr_copied = .false.
+    do i=1,min(max_clen, max_flen)
+      fstr(i:i) = cstr(i)
+      if(fstr(i:i) == C_NULL_CHAR) then
+        fstr(i:i) = F_SPACE_CHAR
+        cstr_copied = .true.
+        exit
+      end if
+    end do
+    if((.not. cstr_copied) .and. (max_flen < max_clen)) then
+      ! We have exhaused the Fortran write buffer
```

Comment:
exhaused ==> exhausted (total 12 occurrences)

Reply:
ok, thanks - will change it